extension/src/popup/helpers/searchAsset.ts

```diff
@@ -113,16 +127,12 @@ export const getVerifiedTokens = async ({
   const verifiedLists: string[] = [];
 
   for (const data of assetListsData) {
-    const list = data.assets;
-    if (list) {
-      for (const record of list) {
-        const regex = new RegExp(contractId, "i");
-        if (record.contract && record.contract.match(regex)) {
-          verifiedToken = record;
-          verifiedLists.push(data.name);
-          break;
-        }
-      }
+    const matchingRecord = data.assets?.find(
+      (record) => record.contract === contractId,
```

Comment:
the above ignores case for the contract ID while the new line explicitly expects the same exact string. We should preserve this to guard against cases where (for ex) a user forgets to capitalize a letter of a contract ID. A very edge case - but unless there's a strong reason to remove this, let's maintain it